src/components/CharacterItem/CharacterItem.test.tsx

```diff
@@ -46,23 +43,45 @@ describe('CharacterItem', () => {
     expect(bmi).not.toBeInTheDocument()
   })
 
+  test('renders properly when no detailed data available', async () => {
+    render(
+      <UnorderedList>
+        <CharacterItem character={withoutBmiCharacter} />
+      </UnorderedList>,
+    )
+
+    const viewDetailsButton = await screen.queryByRole('button', { name: /view details/i })
+    const noDetailsDataMessage = screen.getByText(/no detailed data/i)
+
+    expect(viewDetailsButton).not.toBeInTheDocument()
+    expect(noDetailsDataMessage).toBeInTheDocument()
+  })
+
   test('show details when users click the button and hide them when users click it again', async () => {
     render(
       <UnorderedList>
         <CharacterItem character={withBmiCharacter} />
       </UnorderedList>,
     )
+    // Assign - grab all expected elements to be used in the test
     const viewDetailsButton = screen.getByRole('button', { name: /view details/i })
 
-    // show details
+    // Act - interact by pressing button
     userEvent.click(viewDetailsButton)
-    const height = await screen.findByText(/height/i)
+
+    // Assign
+    const height = await screen.queryByText(/height/i)
+
+    // Assert - make sure the height element is in the screen
     expect(height).toBeInTheDocument()
+
+    // Assign - hide details
     const hideDetailsButton = await screen.getByRole('button', { name: /hide details/i })
 
-    // hide details
+    // Act
     userEvent.click(hideDetailsButton)
-    const noHeight = await screen.queryByText(/height/i)
-    expect(noHeight).not.toBeInTheDocument()
+
+    // Assert - no hight button
```

Comment:
[nitpick] - This is my personal preference so you won't need to follow it, but I'd rather not write comments such as `assign act assert` as it can be quite noisy. You can see where's assigned, act, and assert as testing lib names each API quite declaratively so I won't mind seeing tests without comments that suggest which role each line is, but the good intention to explain the test with comments like `make sure height element is in the screen`!